conans/client/conan_api.py

```diff
@@ -668,14 +669,45 @@ def users_list(self, remote=None):
 
     @api_method
     def search_recipes(self, pattern, remote=None, case_sensitive=False):
+        recorder = SearchRecoder()
         search = Search(self._client_cache, self._remote_manager, self._registry)
-        return search.search_recipes(pattern, remote, case_sensitive)
+
+        try:
+            references = search.search_recipes(pattern, remote, case_sensitive)
+        except ConanException as exc:
+            recorder.error = True
+            exc.info = recorder.get_info()
+            raise
+
+        for remote, refs in references.items():
+            for ref in refs:
+                recorder.add_recipe(str(remote), str(ref[0]), str(ref[1]), with_packages=False)
+        return recorder.get_info()
 
     @api_method
     def search_packages(self, reference, query=None, remote=None, outdated=False):
+        recorder = SearchRecoder()
         search = Search(self._client_cache, self._remote_manager, self._registry)
-        return search.search_packages(reference, remote, query=query,
-                                      outdated=outdated)
+
+        try:
+            ordered_packages, ref, recipe_hash = search.search_packages(reference, remote,
+                                                                        query=query,
+                                                                        outdated=outdated)
+        except ConanException as exc:
+            recorder.error = True
+            exc.info = recorder.get_info()
```

Comment:
Won't get_info() be empty here? What is this for? Doesn't make sense.

Reply:
Yes, but you have to return the error to the excetion output to indicate in the json that the command has failed. You cant try to add information to the recorder as you dont have the return objects of the `search.search_recipe()` call